File: 104-maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int maxDepth(TreeNode root) {
        if (root == null)
            return 0;
        Queue<TreeNode> q = new LinkedList<>();
        q.add(root);
        int max_depth = 0;
        while (!q.isEmpty()){
            int size = q.size();
            max_depth++;
            for (int i = 0; i < size; i++){
                TreeNode temp = q.poll();
                
                if (temp.left != null){
                    q.add(temp.left);
                }
                if (temp.right != null){
                    q.add(temp.right);
                }
            }
        }
        System.out.println(max_depth);
        return max_depth;
    }
}

File: 104-maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxDepth(TreeNode* root) {
        int count1=0;
        int count2=0;
        if(root==nullptr){
            return 0;
        }
        count1++;
        count2++;
        count1+=maxDepth(root->left);
        count2+=maxDepth(root->right);
        return count1>count2?count1:count2;
    }
};